src/handlers/shared/get-user-task-limit-and-role.ts

```diff
@@ -21,11 +21,10 @@ export async function getUserRoleAndTaskLimit(context: Context, user: string): P
       username: user,
     });
 
-    const matchedTask = Object.entries(maxConcurrentTasks).find(
-      ([role]) => role.toLowerCase() === response.data.role.toLowerCase()
-    );
+    const role = response.data.role.toLowerCase();
+    const limit = maxConcurrentTasks[role.charAt(0).toUpperCase() + role.slice(1)];
```

Comment:
It seems you have lowercased `role` and then capitalize it to index `maxConcurrentTasks` in the this line.

We don't know how a partner would define their roles so it makes sense to normalize the octokit returned role and the config roles. Preferably using the `T.Tranform().Decode().Encode()` method mentioned previously and like you have for the octokit response as-is

Reply:
@Keyrxng sorry I'm abit confused on this one ?

Reply:
```ts
const role = response.data.role.toLowerCase();
const limit = maxConcurrentTasks[role.charAt(0).toUpperCase() + role.slice(1)]
```

You make `role` lowercase and then you capitalize the first letter and use it to index `maxConcurrentTasks` which might not match how the partner has defined their roles in the config as they may be lowercase

I'd suggest using the method we discussed [here](https://github.com/ubiquibot/command-start-stop/pull/9#issuecomment-2290370173) to handle decoding or whichever method you think is right

Reply:
I realized I wasn't supposed to do all that gymnastics. I was just supposed to use the octokit returned role and the config role as is.

Reply:
You have defined defaults as being capitalized but how do we guarantee that the role defined in the config by partners is uppercase or lowercase and whatever the octokit response is going to be?

> We don't know how a partner would define their roles so it makes sense to normalize the octokit returned role and the config roles.

Reply:
You're correct that the code as written doesn't enforce consistent casing between the roles defined in the configuration and those returned by octokit. 

To ensure reliable matching between the roles, regardless of how they're cased in the configuration or the API response, we can normalize both the configuration keys and the API response role to a consistent format (e.g., all lowercase) before performing the lookup. Will that work?